tests/unit/amo/components/TestRatingManager.js

```diff
@@ -450,6 +454,80 @@ describe(__filename, () => {
     });
   });
 
+  describe('enableInlineAddonReview', () => {
+    function renderInline(otherProps = {}) {
+      const _config = getFakeConfig({
+        enableInlineAddonReview: true,
+      });
+      return render({ _config, ...otherProps });
+    }
+
+    it('renders a user rating prompt', () => {
+      const root = renderInline();
+
+      expect(root.find(UserRating)).toHaveLength(1);
+      expect(root.find(AddonReviewManager)).toHaveLength(0);
+    });
+
+    it('renders AddonReviewManager after submitting a rating', async () => {
+      const store = createStoreWithLatestReview();
+      const root = renderInline({ store });
+      expect(root.find(AddonReviewManager)).toHaveLength(0);
+
+      await root.instance().onSelectRating(5);
+
+      // Apply setState() changes.
+      expect(root).toHaveState('showTextEntry', true);
+      root.update();
+
+      const manager = root.find(AddonReviewManager);
+      expect(manager).toHaveLength(1);
+      expect(manager).toHaveProp('review', root.instance().props.userReview);
+
+      expect(root.find(UserRating)).toHaveLength(0);
+    });
+
+    it('hides AddonReviewManager when pressing the cancel button', async () => {
+      const root = renderInline({ store: createStoreWithLatestReview() });
+      root.setState({ showTextEntry: true });
+
```

Comment:
Just wondering, why don't we have to call `root.update()` here, whereas we had to in the [previous test](https://github.com/mozilla/addons-frontend/blob/d36773d58b782bd001b53d9cd75c44695d50e5df/tests/unit/amo/components/TestRatingManager.js#L481)?

Reply:
Because `root.setState()` does a re-render internally. In the previous test, the state was changed asynchronously in an event handler which is why `root.update()` was needed. This is actually bad practice because the component could be unmounted which would throw an error (there is a bug about this, too). When we move that component to sagas it can go away.